File: lab02/Curso.java
public class Curso {
    private String nombre;
    private Horario horario;
 
    public Curso(){  
    }
    public Curso (String nombre, String dias, String horaInicio, String horaFin){
        this.nombre=nombre;
        this.horario=new Horario(dias,horaInicio,horaFin);
    }
    public String getNombre(){
        return nombre;
    }
    public void setNombre(String nombre){
        this.nombre=nombre;
    }
    public Horario getHorario(){
        return horario;
    }
    public void setHorario(Horario horario){
        this.horario=horario;
    }
    @Override
    public String toString() {
        return "Curso: " + nombre + " | Horario: " + horario;
    } 
}

File: lab02/Curso.py
from Horario import Horario

class Curso:

    def __init__(
        self,
        nombre: str = "",
        dias: str = "",
        hora_inicio: str = "",
        hora_fin: str = "",
    ):
        self._nombre = nombre
        self._horario = Horario(dias, hora_inicio, hora_fin)

    @property
    def nombre(self) -> str:
        return self._nombre

    @nombre.setter
    def nombre(self, valor: str):
        self._nombre = valor

    @property
    def horario(self) -> Horario:
        return self._horario

    @horario.setter
    def horario(self, valor: Horario):
        self._horario = valor

    def __str__(self) -> str:
        return f"Curso: {self._nombre} | Horario: {self._horario}"